LogMonitor/src/LogMonitor/PipeMonitor.cpp

```diff
@@ -82,14 +114,16 @@ HANDLE CreatePipe() {
     LPCTSTR lpszPipename = TEXT("\\\\.\\pipe\\logMonitor"); 
 
     return CreateNamedPipe(
-        lpszPipename,
-        PIPE_ACCESS_INBOUND,
-        PIPE_TYPE_MESSAGE | PIPE_READMODE_MESSAGE | PIPE_WAIT,   
-        PIPE_UNLIMITED_INSTANCES,
-        512,
-        0,
-        5000,
-        NULL
+        lpszPipename,           // pipe name
+        PIPE_ACCESS_INBOUND,    // read/write access
+        PIPE_TYPE_MESSAGE |    // message type pipe
+        PIPE_READMODE_MESSAGE | // message-read mode
+        PIPE_WAIT,              // blocking mode
+        PIPE_UNLIMITED_INSTANCES, // max. instances
+        512,                        // output buffer size
+        0,                          // input buffer size
+        5000,                       // client time-out
+        NULL                        // default security attribute

```

Comment:
I believe this is what you had in mind with looking into pipe access control and any security risks involved @iankingori


The security descriptor controls access to both client and server ends of the named pipe. If you specify NULL, the named pipe gets a default security descriptor. The ACLs in the default security descriptor for a named pipe grant full control to the LocalSystem account, administrators, and the creator owner. They also grant read access to members of the Everyone group and the anonymous account.

Reply:
I got this good doc on [Creating a Security Descriptor](https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--?redirectedfrom=MSDN) 

Trying to get my head around it.

Reply:
This stack overflow comment also has some insights on this: https://stackoverflow.com/questions/40844884/windows-named-pipe-access-control#:~:text=You%20could%20explicitly,tanking%20the%20CPU.